sdk/servicebus/Azure.Messaging.ServiceBus/src/Compatibility/ServiceBusClientBuilderExtensions.cs

```diff
@@ -1,75 +1,44 @@
-//            where TBuilder : IAzureClientFactoryBuilderWithCredential
-//        {
-//            return builder.RegisterClientFactory<ServiceBusSenderClient, ServiceBusSenderClientOptions>((options, token) => new ServiceBusSenderClient(fullyQualifiedNamespace, eventHubName, token, options));
-//        }
+using Azure.Core.Extensions;
+using Azure.Messaging.ServiceBus;
 
-//        /// <summary>
-//        ///   Registers a <see cref="ServiceBusSenderClient"/> instance with connection options loaded from the provided <paramref name="configuration"/> instance.
-//        /// </summary>
-//        ///
-//        public static IAzureClientBuilder<ServiceBusSenderClient, ServiceBusSenderClientOptions> AddEventHubProducerClient<TBuilder, TConfiguration>(this TBuilder builder, TConfiguration configuration)
-//            where TBuilder : IAzureClientFactoryBuilderWithConfiguration<TConfiguration>
-//        {
-//            return builder.RegisterClientFactory<ServiceBusSenderClient, ServiceBusSenderClientOptions>(configuration);
-//        }
+namespace Microsoft.Extensions.Azure
+{
+    /// <summary>
+    ///    The set of extensions to add the Service Bus client types to the clients builder
+    /// </summary>
+    public static class ServiceBusClientBuilderExtensions
+    {
+        /// <summary>
+        ///   Registers a <see cref="ServiceBusClient "/> instance with the provided <paramref name="connectionString"/>.
+        /// </summary>
+        ///
+        public static IAzureClientBuilder<ServiceBusClient, ServiceBusClientOptions> AddServiceBusClient<TBuilder>(this TBuilder builder, string connectionString)
+            where TBuilder : IAzureClientFactoryBuilder
+        {
+            return builder.RegisterClientFactory<ServiceBusClient, ServiceBusClientOptions>(options => new ServiceBusClient(connectionString, options));
+        }
 
-//        /// <summary>
-//        ///   Registers a <see cref="ServiceBusReceiverClientOptions"/> instance with the provided <paramref name="connectionString"/>
-//        /// </summary>
-//        ///
-//        public static IAzureClientBuilder<ServiceBusReceiverClient, ServiceBusReceiverClientOptions> AddEventHubConsumerClient<TBuilder>(this TBuilder builder, string consumerGroup, string connectionString)
-//            where TBuilder : IAzureClientFactoryBuilder
-//        {
-//            return builder.RegisterClientFactory<ServiceBusReceiverClient, ServiceBusReceiverClientOptions>(options => new ServiceBusReceiverClient(consumerGroup, connectionString, options));
-//        }
+        /// <summary>
+        ///   Registers a <see cref="ServiceBusClient"/> instance with the provided <paramref name="fullyQualifiedNamespace"/>.
+        /// </summary>
+        ///
+        public static IAzureClientBuilder<ServiceBusClient, ServiceBusClientOptions> AddServiceBusClientWithNamespace<TBuilder>(this TBuilder builder, string fullyQualifiedNamespace)
```

Comment:
Why do we use a different method name here instead of an overload?

Reply:
The parameter types are same, that's why we are using different names.